Tests/Runner/RunnerTest.php

```diff
@@ -799,6 +800,446 @@ public function testExecutorStoreStateWithProcessor()
+                ],
+            ],
+        ];
+        $tokenInfo = $this->client->verifyToken();
+        $tokenInfo['owner']['features'] = ['parallel-output-mapping-load'];
+        $clientMock = self::getMockBuilder(Client::class)
+            ->setConstructorArgs([[
+                'url' => STORAGE_API_URL,
+                'token' => STORAGE_API_TOKEN,
+            ]])
+            ->setMethods(['verifyToken'])
+            ->getMock();
+        $clientMock->expects(self::any())
+            ->method('verifyToken')
+            ->will(self::returnValue($tokenInfo));
+        $this->setClientMock($clientMock);
+
+        $runner = $this->getRunner();
+        try {
+            $runner->run(
+                [
+                    new JobDefinition($configData1, new Component($componentData), 'runner-configuration', 'v123', [], 'row-1'),
+                    new JobDefinition($configData2, new Component($componentData), 'runner-configuration', 'v123', [], 'row-2'),
+                ],
+                'run',
+                'run',
+                '1234567',
+                new NullUsageFile()
+            );
+        } catch (UserException $e) {
+            self::assertContains(
+                'Cannot upload file \'out.c-runner-test.my-table-1b.csv\' to table ' .
+                '\'out.c-runner-test.my-table-1b\' in Storage API: There are duplicate columns in CSV file: foo',
+                $e->getMessage()
+            );
+        }
+
+        $component = new Components($this->getClient());
+        $listOptions = new ListConfigurationRowsOptions();
+        $listOptions->setComponentId('keboola.docker-demo-sync')->setConfigurationId('runner-configuration');
+        $configuration = $component->getConfiguration('keboola.docker-demo-sync', 'runner-configuration');
+        // configuration state should be unchanged
+        self::assertArrayHasKey('foo', $configuration['state']);
+        self::assertEquals('bar', $configuration['state']['foo']);
+        $rows = $component->listConfigurationRows($listOptions);
+        uasort(
+            $rows,
+            function ($a, $b) {
+                return strcasecmp($a['id'], $b['id']);
+            }
+        );
+        $row1 = $rows[0];
+        $row2 = $rows[1];
+        self::assertArrayNotHasKey('bazRow1', $row1['state']);
+        self::assertArrayNotHasKey('bazRow2', $row2['state']);
+        self::assertFalse($this->client->tableExists('out.c-runner-test.my-table-1b'));
+        self::assertFalse($this->client->tableExists('out.c-runner-test.my-table-2a'));
+        self::assertFalse($this->client->tableExists('out.c-runner-test.my-table-2b'));
+        self::assertFalse($this->getRunnerHandler()->hasInfoThatContains('Waiting for 4 storage jobs'));
+        $this->clearConfigurations();
+    }
+
+    public function testExecutorStoreStateRowsOutputMappingDeferredLateError()
```

Comment:
tenhle test mi přijde zbytečnej. pokud je někde otestováno, že se output mappingy deferrují a jinde je otestováno, že se state zapisuje až po `waitForStorageJobs`, tak mi přijde zbytečný testovat, jak se to zachová, když je chyba v prvním nebo druhým output mappingu, mělo by to z definice být jedno. a pokud ne, tak je blbě něco jinýho.

Reply:
" jinde je otestováno, že se state zapisuje až po waitForStorageJobs" -> to je prave to, co testuje tenhle test. OM muze vyprodukovat chybu hned - pred zalozenim jobu, nebo az po dokonceni jobu a mezitim je spousta mist, kde by se ten state mohl ulozit a *vypadalo* by to, ze se uklada az po dokonceni OM.  Tohle teda opravdu overuje, ze se state neulozi ani pri te opozdene chybe.